File: runtime/tools/git_metadata.ts
import { tool } from "@opencode-ai/plugin"

export default tool({
  description: "Return only the current target Git branch and commit for report metadata.",
  args: {},
  async execute(_, context) {
    const branch = (await Bun.$`git -C ${context.worktree} branch --show-current`.text()).trim() || "detached"
    const commit = (await Bun.$`git -C ${context.worktree} rev-parse HEAD`.text()).trim()
    return `branch: ${branch}\ncommit: ${commit}`
  },
})

File: runtime/python/analysis_pipeline/tools/git_metadata.py
import subprocess
def git_metadata(worktree):
    def run(*args):
        return subprocess.run(
            ["git", "-C", str(worktree), *args], capture_output=True, text=True, encoding="utf-8", check=True
        ).stdout.strip()
    return f"branch: {run('branch','--show-current') or 'detached'}\ncommit: {run('rev-parse','HEAD')}"
